File: sdk-java/src/main/java/com/crosscert/passkey/sdk/envelope/EnvelopeError.java
package com.crosscert.passkey.sdk.envelope;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;
import java.util.List;

@JsonIgnoreProperties(ignoreUnknown = true)
public record EnvelopeError(
    @JsonProperty("errorCode") String errorCode,
    @JsonProperty("fieldErrors") List<EnvelopeFieldError> fieldErrors
) {}

File: sdk-java/src/main/kotlin/com/crosscert/passkey/sdk/envelope/EnvelopeError.kt
package com.crosscert.passkey.sdk.envelope

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonProperty

@JsonIgnoreProperties(ignoreUnknown = true)
data class EnvelopeError(
    @get:JvmName("errorCode") @get:JsonProperty("errorCode") @param:JsonProperty("errorCode") val errorCode: String?,
    @get:JvmName("fieldErrors") @get:JsonProperty("fieldErrors") @param:JsonProperty("fieldErrors") val fieldErrors: List<EnvelopeFieldError>?,
)
